pkg/apis/cartographer/v1alpha1/workload.go

```diff
@@ -82,9 +82,14 @@ type WorkloadBuild struct {
 }
 
 type WorkloadStatus struct {
-	ObservedGeneration int64              `json:"observedGeneration,omitempty"`
-	Conditions         []metav1.Condition `json:"conditions,omitempty"`
-	SupplyChainRef     ObjectReference    `json:"supplyChainRef,omitempty"`
+	OwnerStatus `json:",inline"`
```

Comment:
What you have is fine, but something to keep in mind is that you can also manipulate the structs a little bit to make things easier to consume, as long as the marshaled yaml is the same.

For example, instead of depending on the inlined type, you could expand it here. That way you don't need to add the extra level of nesting for OwnerStatus in all of the tests.

```suggestion
	// ObservedGeneration refers to the metadata.Generation of the spec that resulted in
 	// the current `status`.
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`

 	// Conditions describing this resource's reconcile state. The top level condition is
 	// of type `Ready`, and follows these Kubernetes conventions:
 	// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Neither approach is right or wrong, but there is a choice.

Reply:
+1